src/couch/src/couch_db.erl

```diff
@@ -742,18 +742,18 @@ is_authorized(#user_ctx{name = UserName, roles = UserRoles}, Security) ->
         false -> check_security(names, UserName, Names)
     end.
 
-check_security(roles, [], _) ->
-    false;
-check_security(roles, UserRoles, Roles) ->
+check_security(roles, [_ | _] = UserRoles, [_ | _] = Roles) ->
     UserRolesSet = ordsets:from_list(UserRoles),
     RolesSet = ordsets:from_list(Roles),
     not ordsets:is_disjoint(UserRolesSet, RolesSet);
-check_security(names, _, []) ->
+check_security(roles, _, _) ->
     false;
+check_security(names, UserName, [_ | _] = Names) ->
```

Comment:
We should put `check_security(names, null, _) ->` before `check_security(names, UserName, [_ | _] = Names)` otherwise we'd still return `true` for `check_security(names, null, [<<"foo">>, null])` which would be new behavior